### DISCUSSION TITLE
Defaut option to git-commit

### DISCUSSION THREAD

**Discussion by chmouel** [magit/magit#4540] [Support] [answered]
Hello, 

Thanks for all the work on magit! I have as tiny question about git-commit mode, 

Is there any way to add a default option to `git-commit-mode`, I would like to add a `"Signed-Off-By"` header but I don't see an option for that.

I have a hook on git-commit-setup which works relatively well : 

```emacs-lisp
(use-package git-commit
  :hook (git-commit-setup . (lambda () (interactive)
                              (save-excursion
                                (goto-char (point-min))
                                (unless (re-search-forward "^Signed-Off-by: " nil t)
                                  (call-interactively 'git-commit-signoff))))))
```

but I would assume this could be a default option somewhere (that i have maybe missed it)

moreover I have this in my gitconfig : 

```init
[alias]
    commit = commit -s
```

but this doesn't seem to take any effect on git-commit-mode

**Comment by tarsius** [comment, marked as answer]
After you have pressed `c` you can enable `--signoff` by pressing `- s`. Since you want to always use that you should then save using `C-x C-s`.

(I recommend against always signing off. IMO if you automatically do it for every commit, then it becomes meaningless and just adds additional noise to the commit message.)

**Reply by psionic-k** [reply]
On revisit, I decided I really want a .dir-locals.el mechanism to add my sign-off for projects using a DCO.

Your argument against automatic assignment is strong.  However, this is a subjective topic, and subjective topics are best left to choice and experimentation.

By exposing a variable for setting via .dir-locals.el, a maintainer can steer the blamelessly oblivious contributor towards noticing, if not the local variable warning, the mysterious Sign-off trailer, both steps that can bring attention to the DCO in use for the project.

Magit could help by giving this trailer a face that stands out over the usual commit comments.
